File: src/main/java/me/interject/math/Intersection.java
package me.interject.math;

public class Intersection {
    public static boolean intersect(int a, int aBounding, int b, int bBounding) {
        int aMin = Math.abs(a - aBounding);
        int aMax = a + aBounding;
        int bMin = Math.abs(b - bBounding);
        int bMax = b + bBounding;

        return (bMin > aMin && bMin < aMax) || (aMin > bMin && aMin < bMax);
    }
}

File: src/main/kotlin/me/interject/math/Intersection.kt
package me.interject.math

/**
 * Created by michaelshaulskiy on 11.09.16.
 */
fun intersect(a: Int, aBounding: Int, b: Int, bBounding: Int): Boolean {
    val aMin = Math.abs(a - aBounding)
    val aMax = a + aBounding
    val bMin = Math.abs(b - bBounding)
    val bMax = b + bBounding

    return (bMin > aMin && bMin < aMax || aMin > bMin && aMin < bMax)
}